### ISSUE TITLE
Huge delays with linting

### ISSUE THREAD

**Issue by Guytoune** [Microsoft/vscode-tslint#287]
For a couple of weeks, tslint is not showing any errors.
Sometimes (I don't know how) it does show the errors but they are never cleared even though I apply a correction.

It works fine when I use the command line.

**Comment by egamma** [comment]
@Guytoune please try this. Open the tslint output log using the command 

TSLint: Show Output. Verify that there is no error message in the shown log. 

Enable tracing output by adding the setting "tslint.trace.server" with a value of "verbose" or "messages" and attach a trace.

**Comment by Guytoune** [comment]
I don't see the tslint warning.
Here is the output :

[Trace - 16:00:55] Sending request 'initialize - (0)'.
Params: {
    "processId": 9264,
    "rootPath": "d:\\Profiles\\glescalier\\Desktop\\vscode-tslint-master\\tslint-tests",
    "rootUri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests",
    "capabilities": {
        "workspace": {
            "applyEdit": true,
            "didChangeConfiguration": {
                "dynamicRegistration": true
            },
            "didChangeWatchedFiles": {
                "dynamicRegistration": true
            },
            "symbol": {
                "dynamicRegistration": true
            },
            "executeCommand": {
                "dynamicRegistration": true
            },
            "workspaceFolders": true,
            "configuration": true
        },
        "textDocument": {
            "synchronization": {
                "dynamicRegistration": true,
                "willSave": true,
                "willSaveWaitUntil": true,
                "didSave": true
            },
            "completion": {
                "dynamicRegistration": true,
                "completionItem": {
                    "snippetSupport": true,
                    "commitCharactersSupport": true
                }
            },
            "hover": {
                "dynamicRegistration": true
            },
            "signatureHelp": {
                "dynamicRegistration": true
            },
            "definition": {
                "dynamicRegistration": true
            },
            "references": {
                "dynamicRegistration": true
            },
            "documentHighlight": {
                "dynamicRegistration": true
            },
            "documentSymbol": {
                "dynamicRegistration": true
            },
            "codeAction": {
                "dynamicRegistration": true
            },
            "codeLens": {
                "dynamicRegistration": true
            },
            "formatting": {
                "dynamicRegistration": true
            },
            "rangeFormatting": {
                "dynamicRegistration": true
            },
            "onTypeFormatting": {
                "dynamicRegistration": true
            },
            "rename": {
                "dynamicRegistration": true
            },
            "documentLink": {
                "dynamicRegistration": true
            }
        }
    },
    "trace": "verbose",
    "workspaceFolders": [
        {
            "uri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests",
            "name": "tslint-tests"
        }
    ]
}


[Trace - 16:00:55] Received response 'initialize - (0)' in 706ms.
Result: {
    "capabilities": {
        "textDocumentSync": 1,
        "codeActionProvider": true
    }
}


[Info  - 16:00:55] Linter is running.
[Trace - 16:00:55] Sending notification 'initialized'.
Params: {}


[Trace - 16:00:55] Sending notification 'workspace/didChangeConfiguration'.
Params: {
    "settings": {
        "tslint": {
            "enable": true,
            "jsEnable": true,
            "rulesDirectory": "",
            "validateWithDefaultConfig": false,
            "configFile": "",
            "ignoreDefinitionFiles": true,
            "exclude": "",
            "run": "onType",
            "nodePath": "",
            "autoFixOnSave": false,
            "alwaysShowRuleFailuresAsWarnings": false,
            "alwaysShowStatus": false,
            "trace": {
                "server": "verbose"
            },
            "packageManager": "npm"
        }
    }
}


[Trace - 16:00:55] Sending notification 'textDocument/didOpen'.
Params: {
    "textDocument": {
        "uri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
        "languageId": "typescript",
        "version": 1,
        "text": "let t: Array<string> = new Array<string>();\n\nconsole.log(t);\n"
    }
}


[Trace - 16:00:56] Received request 'workspace/configuration - (0)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 16:00:56] Sending response 'workspace/configuration - (0)'. Processing request took 2ms
[Trace - 16:00:56] Received request 'workspace/configuration - (1)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 16:00:56] Sending response 'workspace/configuration - (1)'. Processing request took 0ms

**Comment by egamma** [comment]
What version of tslint are you using? 
What is the version of the vscode-tslint extension?

**Comment by Guytoune** [comment]
tslint 5.8.0
vscode-tslint 1.0.14

**Comment by egamma** [comment]
Just tried it with these versions and it works (see below). What output do you see in the output channel?
What version of TypeScript are  you using?

![image](https://user-images.githubusercontent.com/172399/32448404-1b25e910-c30f-11e7-93a7-2b0a7940b41e.png)

**Comment by Guytoune** [comment]
I only have the following in the output :
[Info  - 16:48:18] Linter is running.

I'm using TypeScript 2.6.1.

**Comment by Guytoune** [comment]
I just saw this in the output :
[Info  - 16:56:52] Linter is running.
[Info  - 17:15:57] TSLint library loaded from: d:\Projets\ASF\GeoRepoWaze\GeoRepo.Client.JS.TS\node_modules\tslint\lib\index.js using d:\Profiles\glescalier\AppData\Roaming\npm\node_modules

It's strange that it took 20 minutes to load the library...
It happened before that after being away from my computer, I came back and saw the linter highlights.
It "explains" why when I got the warnings and I correct them, they don't go away.

**Comment by egamma** [comment]
The library is loaded lazily when a file is linted for the first time. The linting is done throttled. I wonder whether an extension can cause the delay. Can you disable the other extensions than tslint and see whether you can reproduce the behaviour.

You can disable extensions in the extensions viewlet.

**Comment by Guytoune** [comment]
Disabling all the extensions but tslint had no effect.

**Comment by Guytoune** [comment]
Just tried unistalling and reinstalling vscode and all extensions (starting with tslint only).
Did not fix my problem.

**Comment by egamma** [comment]
I'll be looking into getting you an instrumented version of the tslint server. It is not clear that you get onDidOpen and why this doesn't trigger a validate sooner.

**Comment by Guytoune** [comment]
Ok. Thanks a lot.
I'll be waiting then :)

**Comment by egamma** [comment]
One more thing. In the tslint-test setup, can you please produce a trace for the following scenario:
- open array-type.ts as before
- type a white space at the end of the file

This is the trace I'm seeing. I'd like to diff it with what you are seing
```
[Trace - 09:09:30] Sending request 'initialize - (0)'.
Params: {
    "processId": 5600,
    "rootPath": "c:\\Users\\egamma\\Projects\\vscode-tslint\\tslint-tests",
    "rootUri": "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests",
    "capabilities": {
        "workspace": {
            "applyEdit": true,
            "didChangeConfiguration": {
                "dynamicRegistration": true
            },
            "didChangeWatchedFiles": {
                "dynamicRegistration": true
            },
            "symbol": {
                "dynamicRegistration": true
            },
            "executeCommand": {
                "dynamicRegistration": true
            },
            "workspaceFolders": true,
            "configuration": true
        },
        "textDocument": {
            "synchronization": {
                "dynamicRegistration": true,
                "willSave": true,
                "willSaveWaitUntil": true,
                "didSave": true
            },
            "completion": {
                "dynamicRegistration": true,
                "completionItem": {
                    "snippetSupport": true,
                    "commitCharactersSupport": true
                }
            },
            "hover": {
                "dynamicRegistration": true
            },
            "signatureHelp": {
                "dynamicRegistration": true
            },
            "definition": {
                "dynamicRegistration": true
            },
            "references": {
                "dynamicRegistration": true
            },
            "documentHighlight": {
                "dynamicRegistration": true
            },
            "documentSymbol": {
                "dynamicRegistration": true
            },
            "codeAction": {
                "dynamicRegistration": true
            },
            "codeLens": {
                "dynamicRegistration": true
            },
            "formatting": {
                "dynamicRegistration": true
            },
            "rangeFormatting": {
                "dynamicRegistration": true
            },
            "onTypeFormatting": {
                "dynamicRegistration": true
            },
            "rename": {
                "dynamicRegistration": true
            },
            "documentLink": {
                "dynamicRegistration": true
            }
        }
    },
    "trace": "verbose",
    "workspaceFolders": [
        {
            "uri": "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests",
            "name": "tslint-tests"
        }
    ]
}


[Trace - 09:09:30] Received response 'initialize - (0)' in 193ms.
Result: {
    "capabilities": {
        "textDocumentSync": 1,
        "codeActionProvider": true
    }
}


[Info  - 09:09:30] Linter is running.
[Trace - 09:09:30] Sending notification 'initialized'.
Params: {}


[Trace - 09:09:30] Sending notification 'workspace/didChangeConfiguration'.
Params: {
    "settings": {
        "tslint": {
            "enable": true,
            "jsEnable": true,
            "rulesDirectory": "",
            "validateWithDefaultConfig": false,
            "configFile": "",
            "ignoreDefinitionFiles": true,
            "exclude": "",
            "run": "onType",
            "nodePath": "",
            "autoFixOnSave": false,
            "alwaysShowRuleFailuresAsWarnings": false,
            "alwaysShowStatus": false,
            "trace": {
                "server": "verbose"
            },
            "packageManager": "npm"
        }
    }
}


[Trace - 09:09:30] Sending notification 'textDocument/didOpen'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests/tests/array-type.ts",
        "languageId": "typescript",
        "version": 1,
        "text": "let t: Array<string> = new Array<string>();\n\nconsole.log(t);\n"
    }
}


[Trace - 09:09:30] Received request 'workspace/configuration - (0)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests/tests/array-type.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 09:09:30] Sending response 'workspace/configuration - (0)'. Processing request took 2ms
[Trace - 09:09:30] Received request 'workspace/configuration - (1)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests/tests/array-type.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 09:09:30] Sending response 'workspace/configuration - (1)'. Processing request took 1ms
[Trace - 09:09:32] 'npm config get prefix' value is: C:\Program Files\nodejs
[Trace - 09:09:34] 'npm config get prefix' value is: C:\Program Files\nodejs
[Trace - 09:09:34] NODE_PATH value is: C:\Program Files\nodejs\node_modules
[Trace - 09:09:35] Received notification 'window/logMessage'.
Params: {
    "type": 3,
    "message": "TSLint library loaded from: c:\\Users\\egamma\\Projects\\vscode-tslint\\tslint-tests\\node_modules\\tslint\\lib\\index.js using C:\\Program Files\\nodejs\\node_modules"
}


[Info  - 09:09:35] TSLint library loaded from: c:\Users\egamma\Projects\vscode-tslint\tslint-tests\node_modules\tslint\lib\index.js using C:\Program Files\nodejs\node_modules
Warning: The 'no-unused-variable' rule requires type information.
[Trace - 09:09:35] tslint configuration:
{ extends: [],
  jsRules: 
   Map {
     'triple-equals' => { ruleArguments: [ 'allow-null-check' ],
     ruleSeverity: 'warning' } },
  linterOptions: {},
  rules: 
   Map {
     'array-type' => { ruleArguments: [ 'array-simple' ], ruleSeverity: 'warning' },
     'arrow-parens' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-var-keyword' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-unused-variable' => { ruleArguments: [ { 'ignore-pattern': '^_' } ],
     ruleSeverity: 'warning' },
     'ordered-imports' => { ruleArguments: 
      [ { 'import-sources-order': 'lowercase-last',
          'named-imports-order': 'lowercase-first' } ],
     ruleSeverity: 'warning' },
     'trailing-comma' => { ruleArguments: [ { multiline: 'always', singleline: 'never' } ],
     ruleSeverity: 'warning' },
     'class-name' => { ruleArguments: [], ruleSeverity: 'warning' },
     'comment-format' => { ruleArguments: [ 'check-space' ], ruleSeverity: 'warning' },
     'indent' => { ruleArguments: [ 'spaces' ], ruleSeverity: 'warning' },
     'no-eval' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-internal-module' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-trailing-whitespace' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-unsafe-finally' => { ruleArguments: [], ruleSeverity: 'warning' },
     'one-line' => { ruleArguments: [ 'check-open-brace', 'check-whitespace' ],
     ruleSeverity: 'warning' },
     'quotemark' => { ruleArguments: [ 'double' ], ruleSeverity: 'warning' },
     'semicolon' => { ruleArguments: [ 'always' ], ruleSeverity: 'warning' },
     'triple-equals' => { ruleArguments: [ 'allow-null-check' ],
     ruleSeverity: 'warning' },
     'typedef-whitespace' => { ruleArguments: 
      [ { 'call-signature': 'nospace',
          'index-signature': 'nospace',
          parameter: 'nospace',
          'property-declaration': 'nospace',
          'variable-declaration': 'nospace' } ],
     ruleSeverity: 'warning' },
     'variable-name' => { ruleArguments: [ 'ban-keywords' ], ruleSeverity: 'warning' },
     'whitespace' => { ruleArguments: 
      [ 'check-branch',
        'check-decl',
        'check-operator',
        'check-separator',
        'check-type' ],
     ruleSeverity: 'warning' } },
  rulesDirectory: [] }
[Trace - 09:09:35] Linting: start linting with tslint > version 4
[Trace - 09:09:35] Linting: ended linting
[Trace - 09:09:35] Received notification 'tslint/status'.
Params: {
    "state": 1
}


[Trace - 09:09:35] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests/tests/array-type.ts",
    "diagnostics": [
        {
            "severity": 2,
            "message": "Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead. (array-type)",
            "range": {
                "start": {
                    "line": 0,
                    "character": 7
                },
                "end": {
                    "line": 0,
                    "character": 20
                }
            },
            "code": "array-type",
            "source": "tslint"
        }
    ]
}


[Trace - 09:09:36] Sending notification 'textDocument/didChange'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests/tests/array-type.ts",
        "version": 2
    },
    "contentChanges": [
        {
            "text": "let t: Array<string> = new Array<string>();\n\nconsole.log(t);\n "
        }
    ]
}


[Trace - 09:09:37] tslint configuration:
{ extends: [],
  jsRules: 
   Map {
     'triple-equals' => { ruleArguments: [ 'allow-null-check' ],
     ruleSeverity: 'warning' } },
  linterOptions: {},
  rules: 
   Map {
     'array-type' => { ruleArguments: [ 'array-simple' ], ruleSeverity: 'warning' },
     'arrow-parens' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-var-keyword' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-unused-variable' => { ruleArguments: [ { 'ignore-pattern': '^_' } ],
     ruleSeverity: 'warning' },
     'ordered-imports' => { ruleArguments: 
      [ { 'import-sources-order': 'lowercase-last',
          'named-imports-order': 'lowercase-first' } ],
     ruleSeverity: 'warning' },
     'trailing-comma' => { ruleArguments: [ { multiline: 'always', singleline: 'never' } ],
     ruleSeverity: 'warning' },
     'class-name' => { ruleArguments: [], ruleSeverity: 'warning' },
     'comment-format' => { ruleArguments: [ 'check-space' ], ruleSeverity: 'warning' },
     'indent' => { ruleArguments: [ 'spaces' ], ruleSeverity: 'warning' },
     'no-eval' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-internal-module' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-trailing-whitespace' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-unsafe-finally' => { ruleArguments: [], ruleSeverity: 'warning' },
     'one-line' => { ruleArguments: [ 'check-open-brace', 'check-whitespace' ],
     ruleSeverity: 'warning' },
     'quotemark' => { ruleArguments: [ 'double' ], ruleSeverity: 'warning' },
     'semicolon' => { ruleArguments: [ 'always' ], ruleSeverity: 'warning' },
     'triple-equals' => { ruleArguments: [ 'allow-null-check' ],
     ruleSeverity: 'warning' },
     'typedef-whitespace' => { ruleArguments: 
      [ { 'call-signature': 'nospace',
          'index-signature': 'nospace',
          parameter: 'nospace',
          'property-declaration': 'nospace',
          'variable-declaration': 'nospace' } ],
     ruleSeverity: 'warning' },
     'variable-name' => { ruleArguments: [ 'ban-keywords' ], ruleSeverity: 'warning' },
     'whitespace' => { ruleArguments: 
      [ 'check-branch',
        'check-decl',
        'check-operator',
        'check-separator',
        'check-type' ],
     ruleSeverity: 'warning' } },
  rulesDirectory: [] }
[Trace - 09:09:37] Linting: start linting with tslint > version 4
[Trace - 09:09:37] Linting: ended linting
[Trace - 09:09:37] Received notification 'tslint/status'.
Params: {
    "state": 1
}


[Trace - 09:09:37] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests/tests/array-type.ts",
    "diagnostics": [
        {
            "severity": 2,
            "message": "Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead. (array-type)",
            "range": {
                "start": {
                    "line": 0,
                    "character": 7
                },
                "end": {
                    "line": 0,
                    "character": 20
                }
            },
            "code": "array-type",
            "source": "tslint"
        },
        {
            "severity": 2,
            "message": "trailing whitespace (no-trailing-whitespace)",
            "range": {
                "start": {
                    "line": 3,
                    "character": 0
                },
                "end": {
                    "line": 3,
                    "character": 1
                }
            },
            "code": "no-trailing-whitespace",
            "source": "tslint"
        }
    ]
}


[Trace - 09:09:37] Sending request 'textDocument/codeAction - (1)'.
Params: {
    "textDocument": {
        "uri": "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests/tests/array-type.ts"
    },
    "range": {
        "start": {
            "line": 3,
            "character": 0
        },
        "end": {
            "line": 3,
            "character": 1
        }
    },
    "context": {
        "diagnostics": [
            {
                "range": {
                    "start": {
                        "line": 3,
                        "character": 0
                    },
                    "end": {
                        "line": 3,
                        "character": 1
                    }
                },
                "message": "trailing whitespace (no-trailing-whitespace)",
                "severity": 2,
                "code": "no-trailing-whitespace",
                "source": "tslint"
            }
        ]
    }
}


[Trace - 09:09:37] Received response 'textDocument/codeAction - (1)' in 9ms.
Result: [
    {
        "title": "Fix: trailing whitespace",
        "command": "_tslint.applySingleFix",
        "arguments": [
            "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests/tests/array-type.ts",
            2,
            [
                {
                    "range": {
                        "start": {
                            "line": 3,
                            "character": 0
                        },
                        "end": {
                            "line": 3,
                            "character": 1
                        }
                    },
                    "newText": ""
                }
            ]
        ]
    },
    {
        "title": "Fix all auto-fixable problems",
        "command": "_tslint.applyAllFixes",
        "arguments": [
            "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests/tests/array-type.ts",
            2,
            [
                {
                    "range": {
                        "start": {
                            "line": 0,
                            "character": 7
                        },
                        "end": {
                            "line": 0,
                            "character": 13
                        }
                    },
                    "newText": ""
                },
                {
                    "range": {
                        "start": {
                            "line": 0,
                            "character": 19
                        },
                        "end": {
                            "line": 0,
                            "character": 20
                        }
                    },
                    "newText": "[]"
                },
                {
                    "range": {
                        "start": {
                            "line": 3,
                            "character": 0
                        },
                        "end": {
                            "line": 3,
                            "character": 1
                        }
                    },
                    "newText": ""
                }
            ]
        ]
    },
    {
        "title": "Disable rule \"no-trailing-whitespace\"",
        "command": "_tslint.applyDisableRule",
        "arguments": [
            "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests/tests/array-type.ts",
            2,
            [
                {
                    "range": {
                        "start": {
                            "character": 0,
                            "line": 3
                        },
                        "end": {
                            "character": 0,
                            "line": 3
                        }
                    },
                    "newText": "// tslint:disable-next-line:no-trailing-whitespace\n"
                }
            ]
        ]
    },
    {
        "title": "Show documentation for \"no-trailing-whitespace\"",
        "command": "_tslint.showRuleDocumentation",
        "arguments": [
            "file:///c%3A/Users/egamma/Projects/vscode-tslint/tslint-tests/tests/array-type.ts",
            2,
            null,
            "no-trailing-whitespace"
        ]
    }
]
```

**Comment by Guytoune** [comment]
This is what I get in the output if I open the file and type the white space immediatly :

[Trace - 10:09:41] Sending request 'initialize - (0)'.
Params: {
    "processId": 1716,
    "rootPath": "d:\\Profiles\\glescalier\\Desktop\\vscode-tslint-master\\tslint-tests",
    "rootUri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests",
    "capabilities": {
        "workspace": {
            "applyEdit": true,
            "didChangeConfiguration": {
                "dynamicRegistration": true
            },
            "didChangeWatchedFiles": {
                "dynamicRegistration": true
            },
            "symbol": {
                "dynamicRegistration": true
            },
            "executeCommand": {
                "dynamicRegistration": true
            },
            "workspaceFolders": true,
            "configuration": true
        },
        "textDocument": {
            "synchronization": {
                "dynamicRegistration": true,
                "willSave": true,
                "willSaveWaitUntil": true,
                "didSave": true
            },
            "completion": {
                "dynamicRegistration": true,
                "completionItem": {
                    "snippetSupport": true,
                    "commitCharactersSupport": true
                }
            },
            "hover": {
                "dynamicRegistration": true
            },
            "signatureHelp": {
                "dynamicRegistration": true
            },
            "definition": {
                "dynamicRegistration": true
            },
            "references": {
                "dynamicRegistration": true
            },
            "documentHighlight": {
                "dynamicRegistration": true
            },
            "documentSymbol": {
                "dynamicRegistration": true
            },
            "codeAction": {
                "dynamicRegistration": true
            },
            "codeLens": {
                "dynamicRegistration": true
            },
            "formatting": {
                "dynamicRegistration": true
            },
            "rangeFormatting": {
                "dynamicRegistration": true
            },
            "onTypeFormatting": {
                "dynamicRegistration": true
            },
            "rename": {
                "dynamicRegistration": true
            },
            "documentLink": {
                "dynamicRegistration": true
            }
        }
    },
    "trace": "verbose",
    "workspaceFolders": [
        {
            "uri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests",
            "name": "tslint-tests"
        }
    ]
}


[Trace - 10:09:41] Received response 'initialize - (0)' in 348ms.
Result: {
    "capabilities": {
        "textDocumentSync": 1,
        "codeActionProvider": true
    }
}


[Info  - 10:09:41] Linter is running.
[Trace - 10:09:41] Sending notification 'initialized'.
Params: {}


[Trace - 10:09:41] Sending notification 'workspace/didChangeConfiguration'.
Params: {
    "settings": {
        "tslint": {
            "enable": true,
            "jsEnable": true,
            "rulesDirectory": "",
            "validateWithDefaultConfig": false,
            "configFile": "",
            "ignoreDefinitionFiles": true,
            "exclude": "",
            "run": "onType",
            "nodePath": "",
            "autoFixOnSave": false,
            "alwaysShowRuleFailuresAsWarnings": false,
            "alwaysShowStatus": false,
            "trace": {
                "server": "verbose"
            },
            "packageManager": "npm"
        }
    }
}


[Trace - 10:09:41] Sending notification 'textDocument/didOpen'.
Params: {
    "textDocument": {
        "uri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
        "languageId": "typescript",
        "version": 1,
        "text": "let t: Array<string> = new Array<string>();\n\nconsole.log(t);\n"
    }
}


[Trace - 10:09:41] Received request 'workspace/configuration - (0)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 10:09:41] Sending response 'workspace/configuration - (0)'. Processing request took 2ms
[Trace - 10:09:41] Received request 'workspace/configuration - (1)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 10:09:41] Sending response 'workspace/configuration - (1)'. Processing request took 1ms
[Trace - 10:09:47] Sending notification 'textDocument/didChange'.
Params: {
    "textDocument": {
        "uri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
        "version": 2
    },
    "contentChanges": [
        {
            "text": "let t: Array<string> = new Array<string>();\n\nconsole.log(t);\n "
        }
    ]
}

**Comment by Guytoune** [comment]
And if I wait for the librry to load (about 6 minutes), I get the linting working until I switch to another file and get back :

[Trace - 09:40:10] Sending request 'initialize - (0)'.
Params: {
    "processId": 11924,
    "rootPath": "d:\\Profiles\\glescalier\\Desktop\\vscode-tslint-master\\tslint-tests",
    "rootUri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests",
    "capabilities": {
        "workspace": {
            "applyEdit": true,
            "didChangeConfiguration": {
                "dynamicRegistration": true
            },
            "didChangeWatchedFiles": {
                "dynamicRegistration": true
            },
            "symbol": {
                "dynamicRegistration": true
            },
            "executeCommand": {
                "dynamicRegistration": true
            },
            "workspaceFolders": true,
            "configuration": true
        },
        "textDocument": {
            "synchronization": {
                "dynamicRegistration": true,
                "willSave": true,
                "willSaveWaitUntil": true,
                "didSave": true
            },
            "completion": {
                "dynamicRegistration": true,
                "completionItem": {
                    "snippetSupport": true,
                    "commitCharactersSupport": true
                }
            },
            "hover": {
                "dynamicRegistration": true
            },
            "signatureHelp": {
                "dynamicRegistration": true
            },
            "definition": {
                "dynamicRegistration": true
            },
            "references": {
                "dynamicRegistration": true
            },
            "documentHighlight": {
                "dynamicRegistration": true
            },
            "documentSymbol": {
                "dynamicRegistration": true
            },
            "codeAction": {
                "dynamicRegistration": true
            },
            "codeLens": {
                "dynamicRegistration": true
            },
            "formatting": {
                "dynamicRegistration": true
            },
            "rangeFormatting": {
                "dynamicRegistration": true
            },
            "onTypeFormatting": {
                "dynamicRegistration": true
            },
            "rename": {
                "dynamicRegistration": true
            },
            "documentLink": {
                "dynamicRegistration": true
            }
        }
    },
    "trace": "verbose",
    "workspaceFolders": [
        {
            "uri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests",
            "name": "tslint-tests"
        }
    ]
}


[Trace - 09:40:13] Received response 'initialize - (0)' in 2961ms.
Result: {
    "capabilities": {
        "textDocumentSync": 1,
        "codeActionProvider": true
    }
}


[Info  - 09:40:13] Linter is running.
[Trace - 09:40:13] Sending notification 'initialized'.
Params: {}


[Trace - 09:40:13] Sending notification 'workspace/didChangeConfiguration'.
Params: {
    "settings": {
        "tslint": {
            "enable": true,
            "jsEnable": true,
            "rulesDirectory": "",
            "validateWithDefaultConfig": false,
            "configFile": "",
            "ignoreDefinitionFiles": true,
            "exclude": "",
            "run": "onType",
            "nodePath": "",
            "autoFixOnSave": false,
            "alwaysShowRuleFailuresAsWarnings": false,
            "alwaysShowStatus": false,
            "trace": {
                "server": "verbose"
            },
            "packageManager": "npm"
        }
    }
}


[Trace - 09:40:13] Sending notification 'textDocument/didOpen'.
Params: {
    "textDocument": {
        "uri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
        "languageId": "typescript",
        "version": 1,
        "text": "let t: Array<string> = new Array<string>();\n\nconsole.log(t);\n"
    }
}


[Trace - 09:40:13] Received request 'workspace/configuration - (0)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 09:40:14] Sending response 'workspace/configuration - (0)'. Processing request took 28ms
[Trace - 09:40:14] Received request 'workspace/configuration - (1)'.
Params: {
    "items": [
        {
            "scopeUri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
            "section": "tslint"
        }
    ]
}


[Trace - 09:40:14] Sending response 'workspace/configuration - (1)'. Processing request took 1ms
[Trace - 09:43:27] 'npm config get prefix' value is: d:\Profiles\glescalier\AppData\Roaming\npm
[Trace - 09:46:37] 'npm config get prefix' value is: d:\Profiles\glescalier\AppData\Roaming\npm
[Trace - 09:46:37] NODE_PATH value is: d:\Profiles\glescalier\AppData\Roaming\npm\node_modules
[Trace - 09:46:38] Received notification 'window/logMessage'.
Params: {
    "type": 3,
    "message": "TSLint library loaded from: d:\\Profiles\\glescalier\\Desktop\\vscode-tslint-master\\tslint-tests\\node_modules\\tslint\\lib\\index.js using d:\\Profiles\\glescalier\\AppData\\Roaming\\npm\\node_modules"
}


[Info  - 09:46:38] TSLint library loaded from: d:\Profiles\glescalier\Desktop\vscode-tslint-master\tslint-tests\node_modules\tslint\lib\index.js using d:\Profiles\glescalier\AppData\Roaming\npm\node_modules
Warning: The 'no-unused-variable' rule requires type information.
[Trace - 09:46:39] tslint configuration:
{ extends: [],
  jsRules: 
   Map {
     'triple-equals' => { ruleArguments: [ 'allow-null-check' ],
     ruleSeverity: 'warning' } },
  linterOptions: {},
  rules: 
   Map {
     'array-type' => { ruleArguments: [ 'array-simple' ], ruleSeverity: 'warning' },
     'arrow-parens' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-var-keyword' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-unused-variable' => { ruleArguments: [ { 'ignore-pattern': '^_' } ],
     ruleSeverity: 'warning' },
     'ordered-imports' => { ruleArguments: 
      [ { 'import-sources-order': 'lowercase-last',
          'named-imports-order': 'lowercase-first' } ],
     ruleSeverity: 'warning' },
     'trailing-comma' => { ruleArguments: [ { multiline: 'always', singleline: 'never' } ],
     ruleSeverity: 'warning' },
     'class-name' => { ruleArguments: [], ruleSeverity: 'warning' },
     'comment-format' => { ruleArguments: [ 'check-space' ], ruleSeverity: 'warning' },
     'indent' => { ruleArguments: [ 'spaces' ], ruleSeverity: 'warning' },
     'no-eval' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-internal-module' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-trailing-whitespace' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-unsafe-finally' => { ruleArguments: [], ruleSeverity: 'warning' },
     'one-line' => { ruleArguments: [ 'check-open-brace', 'check-whitespace' ],
     ruleSeverity: 'warning' },
     'quotemark' => { ruleArguments: [ 'double' ], ruleSeverity: 'warning' },
     'semicolon' => { ruleArguments: [ 'always' ], ruleSeverity: 'warning' },
     'triple-equals' => { ruleArguments: [ 'allow-null-check' ],
     ruleSeverity: 'warning' },
     'typedef-whitespace' => { ruleArguments: 
      [ { 'call-signature': 'nospace',
          'index-signature': 'nospace',
          parameter: 'nospace',
          'property-declaration': 'nospace',
          'variable-declaration': 'nospace' } ],
     ruleSeverity: 'warning' },
     'variable-name' => { ruleArguments: [ 'ban-keywords' ], ruleSeverity: 'warning' },
     'whitespace' => { ruleArguments: 
      [ 'check-branch',
        'check-decl',
        'check-operator',
        'check-separator',
        'check-type' ],
     ruleSeverity: 'warning' } },
  rulesDirectory: [] }
[Trace - 09:46:39] Linting: start linting with tslint > version 4
[Trace - 09:46:39] Linting: ended linting
[Trace - 09:46:39] Received notification 'tslint/status'.
Params: {
    "state": 1
}


[Trace - 09:46:39] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
    "diagnostics": [
        {
            "severity": 2,
            "message": "Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead. (array-type)",
            "range": {
                "start": {
                    "line": 0,
                    "character": 7
                },
                "end": {
                    "line": 0,
                    "character": 20
                }
            },
            "code": "array-type",
            "source": "tslint"
        }
    ]
}


[Trace - 10:03:57] Sending notification 'textDocument/didChange'.
Params: {
    "textDocument": {
        "uri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
        "version": 2
    },
    "contentChanges": [
        {
            "text": "let t: Array<string> = new Array<string>();\n\nconsole.log(t);\n "
        }
    ]
}


[Trace - 10:03:57] tslint configuration:
{ extends: [],
  jsRules: 
   Map {
     'triple-equals' => { ruleArguments: [ 'allow-null-check' ],
     ruleSeverity: 'warning' } },
  linterOptions: {},
  rules: 
   Map {
     'array-type' => { ruleArguments: [ 'array-simple' ], ruleSeverity: 'warning' },
     'arrow-parens' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-var-keyword' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-unused-variable' => { ruleArguments: [ { 'ignore-pattern': '^_' } ],
     ruleSeverity: 'warning' },
     'ordered-imports' => { ruleArguments: 
      [ { 'import-sources-order': 'lowercase-last',
          'named-imports-order': 'lowercase-first' } ],
     ruleSeverity: 'warning' },
     'trailing-comma' => { ruleArguments: [ { multiline: 'always', singleline: 'never' } ],
     ruleSeverity: 'warning' },
     'class-name' => { ruleArguments: [], ruleSeverity: 'warning' },
     'comment-format' => { ruleArguments: [ 'check-space' ], ruleSeverity: 'warning' },
     'indent' => { ruleArguments: [ 'spaces' ], ruleSeverity: 'warning' },
     'no-eval' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-internal-module' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-trailing-whitespace' => { ruleArguments: [], ruleSeverity: 'warning' },
     'no-unsafe-finally' => { ruleArguments: [], ruleSeverity: 'warning' },
     'one-line' => { ruleArguments: [ 'check-open-brace', 'check-whitespace' ],
     ruleSeverity: 'warning' },
     'quotemark' => { ruleArguments: [ 'double' ], ruleSeverity: 'warning' },
     'semicolon' => { ruleArguments: [ 'always' ], ruleSeverity: 'warning' },
     'triple-equals' => { ruleArguments: [ 'allow-null-check' ],
     ruleSeverity: 'warning' },
     'typedef-whitespace' => { ruleArguments: 
      [ { 'call-signature': 'nospace',
          'index-signature': 'nospace',
          parameter: 'nospace',
          'property-declaration': 'nospace',
          'variable-declaration': 'nospace' } ],
     ruleSeverity: 'warning' },
     'variable-name' => { ruleArguments: [ 'ban-keywords' ], ruleSeverity: 'warning' },
     'whitespace' => { ruleArguments: 
      [ 'check-branch',
        'check-decl',
        'check-operator',
        'check-separator',
        'check-type' ],
     ruleSeverity: 'warning' } },
  rulesDirectory: [] }
[Trace - 10:03:57] Linting: start linting with tslint > version 4
[Trace - 10:03:57] Linting: ended linting
[Trace - 10:03:57] Received notification 'tslint/status'.
Params: {
    "state": 1
}


[Trace - 10:03:57] Received notification 'textDocument/publishDiagnostics'.
Params: {
    "uri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
    "diagnostics": [
        {
            "severity": 2,
            "message": "Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead. (array-type)",
            "range": {
                "start": {
                    "line": 0,
                    "character": 7
                },
                "end": {
                    "line": 0,
                    "character": 20
                }
            },
            "code": "array-type",
            "source": "tslint"
        },
        {
            "severity": 2,
            "message": "trailing whitespace (no-trailing-whitespace)",
            "range": {
                "start": {
                    "line": 3,
                    "character": 0
                },
                "end": {
                    "line": 3,
                    "character": 1
                }
            },
            "code": "no-trailing-whitespace",
            "source": "tslint"
        }
    ]
}


[Trace - 10:03:57] Sending request 'textDocument/codeAction - (1)'.
Params: {
    "textDocument": {
        "uri": "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts"
    },
    "range": {
        "start": {
            "line": 3,
            "character": 0
        },
        "end": {
            "line": 3,
            "character": 1
        }
    },
    "context": {
        "diagnostics": [
            {
                "range": {
                    "start": {
                        "line": 3,
                        "character": 0
                    },
                    "end": {
                        "line": 3,
                        "character": 1
                    }
                },
                "message": "trailing whitespace (no-trailing-whitespace)",
                "severity": 2,
                "code": "no-trailing-whitespace",
                "source": "tslint"
            }
        ]
    }
}


[Trace - 10:03:57] Received response 'textDocument/codeAction - (1)' in 4ms.
Result: [
    {
        "title": "Fix: trailing whitespace",
        "command": "_tslint.applySingleFix",
        "arguments": [
            "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
            2,
            [
                {
                    "range": {
                        "start": {
                            "line": 3,
                            "character": 0
                        },
                        "end": {
                            "line": 3,
                            "character": 1
                        }
                    },
                    "newText": ""
                }
            ]
        ]
    },
    {
        "title": "Fix all auto-fixable problems",
        "command": "_tslint.applyAllFixes",
        "arguments": [
            "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
            2,
            [
                {
                    "range": {
                        "start": {
                            "line": 0,
                            "character": 7
                        },
                        "end": {
                            "line": 0,
                            "character": 13
                        }
                    },
                    "newText": ""
                },
                {
                    "range": {
                        "start": {
                            "line": 0,
                            "character": 19
                        },
                        "end": {
                            "line": 0,
                            "character": 20
                        }
                    },
                    "newText": "[]"
                },
                {
                    "range": {
                        "start": {
                            "line": 3,
                            "character": 0
                        },
                        "end": {
                            "line": 3,
                            "character": 1
                        }
                    },
                    "newText": ""
                }
            ]
        ]
    },
    {
        "title": "Disable rule \"no-trailing-whitespace\"",
        "command": "_tslint.applyDisableRule",
        "arguments": [
            "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
            2,
            [
                {
                    "range": {
                        "start": {
                            "character": 0,
                            "line": 3
                        },
                        "end": {
                            "character": 0,
                            "line": 3
                        }
                    },
                    "newText": "// tslint:disable-next-line:no-trailing-whitespace\n"
                }
            ]
        ]
    },
    {
        "title": "Show documentation for \"no-trailing-whitespace\"",
        "command": "_tslint.showRuleDocumentation",
        "arguments": [
            "file:///d%3A/Profiles/glescalier/Desktop/vscode-tslint-master/tslint-tests/tests/array-type.ts",
            2,
            null,
            "no-trailing-whitespace"
        ]
    }
]

**Comment by Guytoune** [comment]
I need insiders version of code ?
I don't have "code-insiders" in my command line and I can't see anything insiders related in code installation directory.

**Comment by egamma** [comment]
Sorry, this also works with the stable VS Code version. (the insiders version is a separate install https://code.visualstudio.com/insiders/).

**Comment by Worlor** [comment]
Hello

I got the same problem than Guy, the linting are very very slow : 


[Trace - 12:20:19] Received response 'textDocument/codeAction - (15)' in 141938ms.

**Comment by egamma** [comment]
@Worlor to help track this down can you please follow the steps described in this comment https://github.com/Microsoft/vscode-tslint/issues/287#issuecomment-342773373 to produce a detailed trace.

**Comment by Worlor** [comment]
I am unable to install the extension with the command line you provide, both with code and code-insiders : VSCode launch but the extensions don't appear in extension list.

**Comment by egamma** [comment]
@Worlor can you make sure that you quit VS Code before you start it with the install extension command line?

**Comment by Worlor** [comment]
Anyway, I give you a trace with my current config : 

- VSCode 1.18.0
- TSLint 1.0.14
- TSC 2.6.1
- Node v8.9.1
- NPM 5.5.1
- Tslint (installed globally with npm) 5.8.0

Tslint.json, file tested and trace attached
[trace-tslint.txt](https://github.com/Microsoft/vscode-tslint/files/1458406/trace-tslint.txt)
[test.ts.txt](https://github.com/Microsoft/vscode-tslint/files/1458409/test.ts.txt)
[tslint.json.txt](https://github.com/Microsoft/vscode-tslint/files/1458412/tslint.json.txt)





The linting appear after a while, but took several minutes :tired_face: 

Thank you

**Comment by Worlor** [comment]
Yes it was well closed

**Comment by egamma** [comment]
@Worlor thanks! There is indeed almost 60s delay from the notification that the text document has changed until the linting started.

I've published a new version of vscode-tslint 1.0.15 which has more trace information. In particular for the 60s where I don't see what is happening on your machine. Can you please update to this version (no need to install a VSIX) and produce and attach another trace.

**Comment by Worlor** [comment]
Ok, I was able to install the extension you provide (1.0.16) with my VSCode-insider. The command line arguments didn't work :confused: :question: 
I used "Install Extension from VSIX" in the "Ctrl + Shift + P" dialog.

Here is the trace : 
[new-trace.txt](https://github.com/Microsoft/vscode-tslint/files/1458511/new-trace.txt)

**Comment by egamma** [comment]
Our messages have crossed, thanks

**Comment by Worlor** [comment]
Yes, thanks anyway :+1:

**Comment by egamma** [comment]
We are getting close. There is more than 1 minute lost here

>[Trace - 18:05:01] SettingsCache fetch for file:///x%3A/SeleniumStarter/src/test.ts
>[Trace - 18:06:05] triggerValidation on file:///x%3A/SeleniumStarter/src/test.ts

I've added more instrumentation in this area. Can you please produce another trace with the vscode-tslint version I've just published (1.0.16)

**Comment by Worlor** [comment]
Same link as above ?

**Comment by egamma** [comment]
it is in the market place, just update the extension. suggest to uninstall your current one.

Am 09.11.2017 um 18:45 schrieb Kentintamarre <notifications@github.com<mailto:notifications@github.com>>:


Same link as above ?

**Comment by Worlor** [comment]
Yes, this is what I have done

(trace)[https://github.com/Microsoft/vscode-tslint/files/1458669/trace-0.16.txt]

**Comment by Swellenator** [comment]
I'd just like to chime in and say I have the same problem. 

```
[Trace - 08:58:34] Sending notification 'textDocument/didChange'.
[Trace - 09:01:40] 'npm config get prefix' value is: C:\Users\[user]\AppData\Roaming\npm
[Trace - 09:01:40] tslint configuration:
[Trace - 09:01:40] Linting: start linting with tslint > version 4
[Trace - 09:01:40] Linting: ended linting
[Trace - 09:01:40] Received notification 'tslint/status'.
[Trace - 09:01:40] Received notification 'textDocument/publishDiagnostics'.
```

There is a long wait between the did change and the next step. I can do the detailed trace with the other version if you want.

**Comment by egamma** [comment]
Thanks for all the traces! 

There was a bug in that the expensive function to compute the path to the global module directory was called too often. The call is now properly cached and should happen only once.

I've published the fix in 1.0.17 to the marketplace. Please verify whether the caching is sufficient to address the delay.

**Comment by Worlor** [comment]
Thanks for the update

The first linting has take ~30-40s  but it's working good now :+1:

**Comment by egamma** [comment]
> The first linting has take ~30-40s but it's working good now

I Still need to investigate why computing the global node module path is that slow on some machines. I cannot reproduce this on my machine.

**Comment by egamma** [comment]
@Worlor first can you please produce another trace with vscode-tslint 1.0.18 that I just published.

Can you think of anything that is special about your node setup. From the trace it looks like your code is on an `x:` drive. Is this a network share?

**Comment by Worlor** [comment]
Can I help you to resolve this ? Tell me if you need some traces :wink:

**Comment by Guytoune** [comment]
With the last version it works fine past the first load.
This is the from the output :

[Trace - 11:01:30] resolveGlobalNodePath: start
[Trace - 11:04:43] 'npm config get prefix' value is: d:\Profiles\glescalier\AppData\Roaming\npm
[Trace - 11:04:43] resolveGlobalNodePath: done

It's like you said. The resolveGlobalNodePath function takes 3 minutes on my computer...
The path is local and  I have a SSD.

**Comment by Worlor** [comment]
Sure, here it is
[trace-0.18.txt](https://github.com/Microsoft/vscode-tslint/files/1460938/trace-0.18.txt)

X: is a local virtual drive mapped to a folder in C: -> Enterprise rules

FYI : Traces coming from Code 1.18

**Comment by egamma** [comment]
Thanks!

So the traces confirm that the time is spent in the computation of the location of the global node path.
This function spawns a process with the following command

`npm config get prefix` 

Can you please run this command on the command line and tell me how long it took.

**Comment by egamma** [comment]
Found this issue https://github.com/npm/npm/issues/14458 `npm config get prefix` takes incredibly long (7 - 70 seconds)

**Comment by egamma** [comment]
I've changed the implementation so that the global node path is accessed more lazily (published in 1.0.19). This enables the following work arounds for the slow down:
- install tslint locally
- define the setting `tslint.nodePath` to point to your node installation

**Comment by Worlor** [comment]
Thanks for your work

I updated the extension and defined the settings, but the first launch is still slow.
The npm command is working properly
![2017-11-10_14-18-17](https://user-images.githubusercontent.com/3814031/32660374-d7348baa-c622-11e7-8fc6-e71d8e52db3e.gif)

Here is a new trace : 
[trace-0.19.txt](https://github.com/Microsoft/vscode-tslint/files/1461462/trace-0.19.txt)

**Comment by egamma** [comment]
Interesting, but this comment https://github.com/npm/npm/issues/14458#issuecomment-257019817 hints at the command is not always slow.

I'll do some more digging.

**Comment by Guytoune** [comment]
I executed the command a couple times and it never took more than 2 seconds.

**Comment by Worlor** [comment]
Strange,
I have tried the powershell command Measure-Time to see the time "npm config get prefix" took.
![powershell_2017-11-10_15-21-57](https://user-images.githubusercontent.com/3814031/32662488-04633362-c62b-11e7-9b3d-74704d9ea83e.png)

If I just type the npm command it's take < 1s, but with Measure-Time it's slow :flushed:

**Comment by egamma** [comment]
I've now added trace information for  the code that spawns a process to run `npm config get prefix`. It is published in 1.0.20.
```
[Trace - 15:54:46] 'npm config get prefix' starting.
[Trace - 15:54:49] 'npm config get prefix' returned value is: C:\Program Files\nodejs
```
This is the output I'm getting. It takes 3s on my machine which is slower when running from a command prompt.

Can you please do another trace and look for the above trace output.

**Comment by Worlor** [comment]
Thank you, here is the trace
[trace-0.20.txt](https://github.com/Microsoft/vscode-tslint/files/1461774/trace-0.20.txt)

The bug come from npm, I will try to change version.

**Comment by egamma** [comment]
The trace shows that the call to `npm config get prefix. Takes over a minute.

Upgrading to a newer version of npm is a good idea. I'm using v4.2.0 of npm. However, the issue https://github.com/npm/npm/issues/14458 is still open, but it is still worth a try.

```
[Trace - 16:15:44] 'npm config get prefix' starting.
[Trace - 16:16:49] 'npm config get prefix' returned value is: C:\Program Files\nodejs
```

**Comment by Worlor** [comment]
In fact, I was on the last version of npm, I downgraded to v4.2 and it's fast now (3s)

However, the "tslint.nodepath" config has no effect

Trace related
[trace-npm-4.2.txt](https://github.com/Microsoft/vscode-tslint/files/1461932/trace-npm-4.2.txt)

**Comment by Guytoune** [comment]
After downgrading npm to 5.4 it worked.
Then I trie updating it again to last version (5.5.1) and I dont't seem to have the issue any more.
Weird...

**Comment by egamma** [comment]
> However, the "tslint.nodepath" config has no effect

Issue confirmed, this has been fixed with 7e8717e86116b395ee51cb13b4c240b56586ad4b and published in v1.0.21.

**Comment by egamma** [comment]
The call to the (sometimes) expensive function that invokes `npm config get prefix` is now deferred as much as possible. It is now only called when tslint is installed globally and the setting `tslint.nodePath` has **not** been set. 

I cannot do more inside the extension to work around this issue and I'm closing the issue as an upstream npm issue.

Thanks to @Guytoune and @Worlor for helping me track down this performance problem.

**Comment by OiYouYeahYou** [comment]
Could the advice of trying different node versions be added to the FAQ. I've wasted a lot of time fixing this one problem. :(

**Comment by egamma** [comment]
@OiYouYeahYou very good point. 

Can you please tell me what node versions worked/ did not work for you?

What I know:
npm 4.2.0 OK
npm 5.4 OK
npm 5.5.1 OK/NOK see comment above. 
what version did cause problems for you?

The workaround is to define tslint.nodepath in the settings or to install tslint locally. 

May be the best way to handle this is to describe the workarounds since it is difficult to narrow down the npm version with the problem.

**Comment by OiYouYeahYou** [comment]
I was on node 7.8.0, which led to it being practically dead for most of the time.
Changed to 4.2.0, now it responds within five seconds, mostly a second or two

**Comment by egamma** [comment]
I've added an item to the FAQ section that documents how to work around this issue.
